components/script/dom/performancetiming.rs

```diff
@@ -0,0 +1,282 @@
+    domain_lookup_end: Cell<u64>,
+    connect_start: Cell<u64>,
+    connect_end: Cell<u64>,
+    secure_connection_start: Cell<u64>,
+    request_start: Cell<u64>,
+    response_start: Cell<u64>,
+    response_end: Cell<u64>,
+    dom_loading: Cell<u64>,
+    dom_interactive: Cell<u64>,
+    dom_content_loaded_event_start: Cell<u64>,
+    dom_content_loaded_event_end: Cell<u64>,
+    dom_complete: Cell<u64>,
+    load_event_start: Cell<u64>,
+    load_event_end: Cell<u64>,
+}
+
+#[derive(Default)]
+pub struct PerformanceTimingBuilder {
+    navigation_start: u64,
+    unload_event_start: u64,
+    unload_event_end: u64,
+    redirect_start: u64,
+    redirect_end: u64,
+    fetch_start: u64,
+    domain_lookup_start: u64,
+    domain_lookup_end: u64,
+    connect_start: u64,
+    connect_end: u64,
+    secure_connection_start: u64,
+    request_start: u64,
+    response_start: u64,
+    response_end: u64,
+    dom_loading: u64,
+    dom_interactive: u64,
+    dom_content_loaded_event_start: u64,
+    dom_content_loaded_event_end: u64,
+    dom_complete: u64,
+    load_event_start: u64,
+    load_event_end: u64,
+}
+
+#[allow(dead_code)]
+impl PerformanceTiming {
+    pub fn new_inherited(config: PerformanceTimingBuilder) -> PerformanceTiming {
+        PerformanceTiming {
+            reflector_: Reflector::new(),
+            navigation_start: Cell::new(config.navigation_start),
+            unload_event_start: Cell::new(config.unload_event_start),
+            unload_event_end: Cell::new(config.unload_event_end),
+            redirect_start: Cell::new(config.redirect_start),
+            redirect_end: Cell::new(config.redirect_end),
+            fetch_start: Cell::new(config.fetch_start),
+            domain_lookup_start: Cell::new(config.domain_lookup_start),
+            domain_lookup_end: Cell::new(config.domain_lookup_end),
+            connect_start: Cell::new(config.connect_start),
+            connect_end: Cell::new(config.connect_end),
+            secure_connection_start: Cell::new(config.secure_connection_start),
+            request_start: Cell::new(config.request_start),
+            response_start: Cell::new(config.response_start),
+            response_end: Cell::new(config.response_end),
+            dom_loading: Cell::new(config.dom_loading),
+            dom_interactive: Cell::new(config.dom_interactive),
+            dom_content_loaded_event_start: Cell::new(config.dom_content_loaded_event_start),
+            dom_content_loaded_event_end: Cell::new(config.dom_content_loaded_event_end),
+            dom_complete: Cell::new(config.dom_complete),
+            load_event_start: Cell::new(config.load_event_start),
+            load_event_end: Cell::new(config.load_event_end),
+        }
+    }
+
+    pub fn new(global: &GlobalScope, navigation_start: u64) -> DomRoot<PerformanceTiming> {
+        let config = PerformanceTimingBuilder {
+            navigation_start,
+            ..Default::default()
+        };
```

Comment:
This builder type feels unnecessary right now, since we only pass in one value and there are no other users of it. Can we remove it and just set the field directly?